alphastats/gui/pages/03_Preprocessing.py

```diff
@@ -1,16 +1,88 @@
 import streamlit as st
+import pandas as pd
 
-try:
-    from alphastats.gui.utils.preprocessing_helper import main_preprocessing
-    from alphastats.gui.utils.ui_helper import sidebar_info
 
-except ModuleNotFoundError:
-    from utils.preprocessing_helper import main_preprocessing
-    from utils.ui_helper import sidebar_info
+from alphastats.gui.utils.preprocessing_helper import (
+    draw_predefined_workflow,
+    configure_preprocessing,
+    update_workflow,
+    run_preprocessing,
+    reset_preprocessing,
+)
+from alphastats.gui.utils.ui_helper import sidebar_info
 
 
 sidebar_info()
 
+if "workflow" not in st.session_state:
+    st.session_state.workflow = [
+        "remove contaminations",
+        "subset data",
+        "log2 transform",
+    ]
+
 st.markdown("### Preprocessing")
+st.markdown(
+    "Select either the predefined workflow where you can only enable/disable steps or create a custom workflow, that allows switching steps around."
+)
+
+
+tab1, tab2 = st.tabs(["Predefined workflow", "Custom workflow"])
+
+with tab1:
+    c1, c2 = st.columns([1, 1])
+
+    with c2:
+        (
+            remove_contaminations,
+            remove_samples,
+            subset,
+            data_completeness,
+            log2_transform,
+            normalization,
+            imputation,
+            batch,
+        ) = configure_preprocessing()
```

Comment:
what if we introduce a dictionary that maps all preprocessing steps to their parameters?

```python
mapping = {
"imputation": "median",
"log2_transform": False,
"normalization": None
...
}
```
this has several benefits:
- only one argument is passed around
- order of imputations can be encoded
- serialization is easier

Reply:
I would go straight for returning a workflow object, once this is designed (https://github.com/MannLabs/alphapeptstats/pull/308#discussion_r1751477966)

Reply:
https://github.com/MannLabs/alphapeptstats/pull/308/commits/4dad21d8017ebb8e4574f1b564d71c7dba1c3ad6

Reply:
https://github.com/MannLabs/alphapeptstats/pull/308/commits/f477837bf9467165232ff7db4c17e7880664f92b

now also using the dictionary